activerecord/lib/active_record/attribute_methods/dirty.rb

```diff
@@ -168,11 +165,15 @@ def write_attribute_without_type_cast(attr_name, _)
         end
 
         def _update_record(*)
-          partial_writes? ? super(keys_for_partial_write) : super
+          affected_rows = partial_writes? ? super(keys_for_partial_write) : super
+          changes_applied
+          affected_rows
```

Comment:
I'm prefer not to care about what kind of value is returned in this module.
Can you use `ensure` instead?

```diff
diff --git a/activerecord/lib/active_record/attribute_methods/dirty.rb b/activerecord/lib/active_record/attribute_methods/dirty.rb
index 233ee29fac..58613f5e88 100644
--- a/activerecord/lib/active_record/attribute_methods/dirty.rb
+++ b/activerecord/lib/active_record/attribute_methods/dirty.rb
@@ -169,10 +169,14 @@ def write_attribute_without_type_cast(attr_name, _)

         def _update_record(*)
           partial_writes? ? super(keys_for_partial_write) : super
+        ensure
+          changes_applied
         end

         def _create_record(*)
           partial_writes? ? super(keys_for_partial_write) : super
+        ensure
+          changes_applied
         end

         def keys_for_partial_write
```

Reply:
hm... I've confirmed using `ensure` would change current behavior if any exception is raised from database.
We should not use `ensure` at least in this point.